### DISCUSSION TITLE
Draft Pages

### DISCUSSION THREAD

**Discussion by richdynamix** [Z3d0X/filament-fabricator#40] [Ideas]
Is it possible to allow draft pages so the frontend controller does not show a page where its published status is false?

it’s is often useful for admins who create pages to work behind the scenes in draft mode where they can preview the page before pushing it live.

**Comment by Z3d0X** [comment]
Currently its not possible, but this is a good suggestion

**Reply by richdynamix** [reply]
I managed to accomplish this with some changes to the core code. I am happy to submit a PR that would allow this. As an overview, the approach I took was as follows - 

- Created a new `visibility` field within the page migration
- Created a new Form select field with the options of `public` or `private`
- Modified the routes for Fabricator by updating the query for pulling the right page data e.g. - 

```php
$filamentFabricatorPage = FilamentFabricator::getPageModel()::query()
                            ->where('id', $pageId)
                            ->where(function ($query) {
                                $query->where('visibility', 'public')
                                    ->orWhere(function ($query) {
                                        if (Auth::check()) {
                                            $query->where('visibility', 'private');
                                        }
                                    });
                            })
                            ->firstOrFail();
```

Further expansion on this might be to add a conditional field in the form to allow a password to be set which could allow unauthenticated guests to view the page. Something similar to how Wordpress and Statamic handle this feature.
